wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceManager.java

```diff
@@ -58,9 +58,15 @@
   private static final Logger LOG = LoggerFactory.getLogger(WorkspaceManager.class);
 
   /** This attribute describes time when workspace was created. */
-  public static final String CREATED_ATTRIBUTE_NAME = "created";
-  /** This attribute describes time when workspace was last updated or started/stopped/recovered. */
-  public static final String UPDATED_ATTRIBUTE_NAME = "updated";
+  protected static final String CREATED_ATTRIBUTE_NAME = "created";
+  /** This attribute describes time when workspace was last updated or started */
+  protected static final String UPDATED_ATTRIBUTE_NAME = "updated";
+  /** This attribute describes time when workspace was last stopped */
+  protected static final String STOPPED_ATTRIBUTE_NAME = "stopped";
+  /** This attribute indicates is last stop was abnormal */
+  protected static final String STOPPED_ABNORMALLY_ATTRIBUTE_NAME = "stoppedAbnormally";
```

Comment:
Is it used anywhere?
As for me abnormal stop is the case when workspace was running and abnormally stopped (like crashed).
Also when a workspace is running and becomes stopped is failing to start but not abnormal stopping.
WDYT?

Reply:
doesn't get the idea., sorry. We probably can use it when WS is crashed, too.

Reply:
We can but we don't. Let's do it or provide the corresponding comment. Now it's not clear enough.